### PR TITLE
Refactor FXIOS-5808 [v115] updated js injection and retrieval scripts for credit card

### PR DESCRIPTION
**Pull request by nbhasin2**
[Jira ticket - 5808](https://mozilla-hub.atlassian.net/browse/FXIOS-5808)

### Description
Updated injection scripts with latest changes for fathom.js and auto fetching of credit card scripts from HG

TODO: 

- [x] Add ticket to fix injection (fixed in this ticket)

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Review by issammani — APPROVED** [review]
JS side looks good. All the files are there. Tested locally and things work as expected. Thanks !

**Comment by nbhasin2** [comment]
Added a few tests but found out that there is a small issue with the injection. 

We don't have to fix it in this PR but more tests will follow through once the error is resolved.

```TypeError: window.__firefox__.CreditCardHelper.fillCreditCardInfo is not a function. (In 'window.__firefox__.CreditCardHelper.fillCreditCardInfo('{"cc-exp-month":"3","cc-name":"Allen","cc-exp-year":"26","cc-number":"4526373728283737","cc-exp":"3\/26"}')', 'window.__firefox__.CreditCardHelper.fillCreditCardInfo' is undefined)
```

**Comment by nbhasin2** [comment]
Going to re-run, something happened here that it got aborted https://app.bitrise.io/build/de766c8d-7a98-4959-85fd-865be8d0f30c

**Comment by issammani** [comment]
> Added a few tests but found out that there is a small issue with the injection swift script.
> 
> We don't have to fix it in this PR but more tests will follow once the error is resolved.
> 
> ```
> 
> TypeError: window.__firefox__.CreditCardHelper.fillCreditCardInfo is not a function. (In 'window.__firefox__.CreditCardHelper.fillCreditCardInfo('{"cc-exp-month":"3","cc-name":"Allen","cc-exp-year":"26","cc-number":"4526373728283737","cc-exp":"3\/26"}')', 'window.__firefox__.CreditCardHelper.fillCreditCardInfo' is undefined)
> ```

Good catch ! Fixed in [7993058](https://github.com/mozilla-mobile/firefox-ios/pull/14298/commits/799305850e0611244aae8d4ecfb5c149033daa2f) ( This why tests will be very valuable in the future 😄 )

File: Client/Assets/CC_Script/FormAutofill.sys.mjs [hunk 1]
```diff
@@ -0,0 +1,270 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
```
Comment (adudenamedruby) [inline Client/Assets/CC_Script/FormAutofill.sys.mjs:4]: I keep seeing this format for these headers. Is Swiftlint not screaming?

File: Client/Frontend/TabContentsScripts/CreditCardHelper.swift [hunk 2]
```diff
@@ -8,28 +8,31 @@ import WebKit
...
 import Storage
 
-enum ValidFieldType: String, CaseIterable {
-    case ccNumber = "cc-number"
-    case ccExpMonth = "cc-exp-month"
-    case ccExpYear = "cc-exp-year"
-    case ccName = "cc-name"
-}
-
 struct Payload: Codable {
```
Comment (adudenamedruby) [inline Client/Frontend/TabContentsScripts/CreditCardHelper.swift:11]: Would something like `CreditCardPayload` be a better name for this? Then you don't have to prefix with `cc` for stuff, and it reads more straightforward? Plus, I'm seeing a bunch of other things with `CreditCard` prefix in the title

File: Client/Frontend/UserContent/UserScripts/AllFrames/AutofillAtDocumentStart/CreditCardHelper.js [hunk 3]
```diff
@@ -1,22 +1,28 @@
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this
-// file, You can obtain one at http://mozilla.org/MPL/2.0/
+// /* This Source Code Form is subject to the terms of the Mozilla Public
+//  * License, v. 2.0. If a copy of the MPL was not distributed with this
+//  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
[the remark this answers is not in the document]
Comment (nbhasin2) [reply Client/Frontend/UserContent/UserScripts/AllFrames/AutofillAtDocumentStart/CreditCardHelper.js:3]: It triggered my personal warning, I thought I fixed it, going to push now

File: Client/Frontend/TabContentsScripts/CreditCardHelper.swift [hunk 4]
```diff
@@ -53,21 +56,28 @@ class CreditCardHelper: TabContentScript {
...
-        var fieldTypes = payload.fieldTypes
-        let fieldTypeValues = getFieldTypeValues(fieldTypes: fieldTypes)
-        foundFieldValues?(fieldTypeValues)
+        foundFieldValues?(getFieldTypeValues(payload: payload))
+//        let plainCreditCard = UnencryptedCreditCardFields(ccName: "Allen Mocktail",
+//                                                          ccNumber: "1234 4567 4567 6788",
+//                                                          ccNumberLast4: "6788",
+//                                                          ccExpMonth: 01,
+//                                                          ccExpYear: 2999, ccType: "Visa")
+//        injectCardInfo(card: plainCreditCard, tab: tab!, completion: {_ in })
```
[the remark this answers is not in the document]
Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/CreditCardHelper.swift:71]: Yes sir, removed it during our initial review just not pushed :)

File: Client/Frontend/TabContentsScripts/CreditCardHelper.swift [hunk 5]
```diff
@@ -53,21 +56,28 @@ class CreditCardHelper: TabContentScript {
...
+    }
+
+    func getValidPayloadData(from message: WKScriptMessage) -> [String: Any]? {
+        return message.body as? [String: Any]
     }
 
     func parseFieldType(messageBody: [String: Any]) -> FillCreditCardForm? {
+        print(String(data: try! JSONSerialization.data(withJSONObject: messageBody, options: .prettyPrinted), encoding: .utf8)!)
```
[the remark this answers is not in the document]
Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/CreditCardHelper.swift:79]: Not at all, removed as well during our initial review. Need to hit push

File: Client/Assets/CC_Script/FormAutofill.sys.mjs (hunk 1 above)
Comment (nbhasin2) [reply Client/Assets/CC_Script/FormAutofill.sys.mjs:4]: Not at all, as this is technically not our code. Its added to XCode so we can easily find and make changes IF required.

Comment (nbhasin2) [reply Client/Assets/CC_Script/FormAutofill.sys.mjs:4]: image

File: Client/Frontend/TabContentsScripts/CreditCardHelper.swift (hunk 2 above)
Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/CreditCardHelper.swift]: Good call, changed name to `CreditCardPayload`

**Comment by nbhasin2** [comment]
image

**Review by adudenamedruby — APPROVED** [review]

**Comment by nbhasin2** [comment]
These failed tests have nothing to do with the changes added here 
image

Going to re-run

**Comment by nbhasin2** [comment]
Something is up with BR - https://app.bitrise.io/build/2f3a9d55-a9e2-4cbf-a283-eb98464b175b?tab=log

It just aborted my build after 70 min. Going to merge this in and start looking into test issues in the epic branch.

Need the Epic branch to pass and have green build before I can merge anyways

**Pull request merged**